ensime_shared/ensime.py

```diff
@@ -279,11 +336,15 @@ def au_buf_leave(self, client, filename):
 
     @execute_with_client()
     def au_cursor_hold(self, client, filename):
-        client.on_cursor_hold(filename)
+        """Handler for CursorHold autocommand event."""
+        self.watchdog_client_server_connection(client)
+        client.unqueue_and_display(filename)
+        client.editor.cursorhold()
```

Comment:
It's isn't our fault this time, we're only the middleman here. The responsibility for the blocking is shared between Ensime server which from time to time likes taking a nap before responding, and the naive omnicompletion API Vim provides (and we hook into) which is called synchronously from (I guess since it blocks) the UI thread.